/*
 *  Copyright (C) 2020-2022 Embedded AMS B.V. - All Rights Reserved
 *
 *  This file is part of Embedded Proto.
 *
 *  Embedded Proto is open source software: you can redistribute it and/or 
 *  modify it under the terms of the GNU General Public License as published 
 *  by the Free Software Foundation, version 3 of the license.
 *
 *  Embedded Proto  is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License
 *  along with Embedded Proto. If not, see <https://www.gnu.org/licenses/>.
 *
 *  For commercial and closed source application please visit:
 *  <https://EmbeddedProto.com/license/>.
 *
 *  Embedded AMS B.V.
 *  Info:
 *    info at EmbeddedProto dot com
 *
 *  Postal address:
 *    Johan Huizingalaan 763a
 *    1066 VH, Amsterdam
 *    the Netherlands
 */

// This file is used to test serializing repeated fields.

syntax = "proto3";

message repeated_fields 
{
  uint32 x            = 1;
  repeated uint32 y   = 2;
  uint32 z            = 3;
}

message repeated_nested_message
{
  uint32 u = 1;
  uint32 v = 2;
}

message repeated_message
{
  uint32 a                            = 1;
  repeated repeated_nested_message b  = 2;
  uint32 c                            = 3;
}

message nested_repeated_message
{
    repeated_fields rf = 1;
}

enum SomeEnum {
  SE_A = 0;
  SE_B = 1;
  SE_C = 2;
}

message repeated_enum {
  repeated SomeEnum enum_values = 1;
}